services/director-v2/src/simcore_service_director_v2/models/schemas/dynamic_services/scheduler.py

```diff
@@ -0,0 +1,442 @@
+        containers_inspect contains some elements.
+        """
+        return self.was_compose_spec_submitted or len(self.containers_inspect) > 0
+
+    was_compose_spec_submitted: bool = Field(
+        False,
+        description="if the docker-compose spec was already submitted this fields is True",
+    )
+
+    containers_inspect: List[DockerContainerInspect] = Field(
+        [],
+        scription="docker inspect results from all the container ran at regular intervals",
+    )
+
+    were_services_created: bool = Field(
+        False,
+        description=(
+            "when True no longer will the Docker api "
+            "be used to check if the services were started"
+        ),
+    )
+
+    @property
+    def can_save_state(self) -> bool:
+        """
+        Keeps track of the current state of the application, if it was starte successfully
+        the state of the service can be saved when stopping the service
+        """
+        # TODO: implement when adding save status hooks
+        return False
+
+    # consider adding containers for healthchecks but this is more difficult and it depends on each service
+
+    @property
+    def endpoint(self):
+        """endpoint where all the services are exposed"""
+        return f"http://{self.hostname}:{self.port}"
+
+    @property
+    def are_containers_ready(self) -> bool:
+        """returns: True if all containers are in running state"""
+        return all(
+            docker_container_inspect.status == DockerStatus.RUNNING
+            for docker_container_inspect in self.containers_inspect
+        )
+
+
+class ServiceLabelsStoredData(CommonServiceDetails, DynamicSidecarServiceLabels):
+    service_name: str
+    paths_mapping: PathMappingsLabel  # overwrites in DynamicSidecarServiceLabels
+    dynamic_sidecar_network_name: str
+    simcore_traefik_zone: str
+    service_port: int
+
+    @classmethod
+    def from_service(cls, service: Dict[str, Any]) -> "ServiceLabelsStoredData":
+        labels = service["Spec"]["Labels"]
+        params = dict(
+            service_name=service["Spec"]["Name"],
+            node_uuid=NodeID(labels["uuid"]),
+            key=labels["service_key"],
+            version=labels["service_tag"],
+            paths_mapping=PathMappingsLabel.parse_raw(labels["paths_mapping"]),
+            dynamic_sidecar_network_name=labels["traefik.docker.network"],
+            simcore_traefik_zone=labels["io.simcore.zone"],
+            service_port=labels["service_port"],
+            project_id=ProjectID(labels["study_id"]),
+            user_id=int(labels["user_id"]),
+        )
+        if "compose_spec" in labels:
+            params["compose_spec"] = json.loads(labels["compose_spec"])
+        if "container_http_entry" in labels:
+            params["container_http_entry"] = labels["container_http_entry"]
+
+        return cls(**params)
+
+    class Config:
+        extra = Extra.allow
```

Comment:
why allow here?

Reply:
Because it is inheriting from `DynamicSidecarServiceLabels` which is configured as `Extra.forid` and is not able to merge extra arguments coming from other models via inheritance and additional properties.